src/main/java/blue/lapis/pore/impl/PoreIpBanList.java

```
@@ -0,0 +1,72 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package blue.lapis.pore.impl;
+
+import org.spongepowered.api.util.ban.Ban;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Collection;
+import java.util.Optional;
+
+public final class PoreIpBanList extends PoreBanList<Ban.Ip> {
+
+    @Override
+    protected Optional<Ban.Ip> getBanFor(String target) {
+        try {
+            return getBanService().getBanFor(InetAddress.getByName(target));
+        } catch (UnknownHostException e) {
+            return Optional.empty();
+        }
+    }
+
+    @Override
+    protected void setTarget(Ban.Builder builder, String target) {
+        try {
+            builder.address(InetAddress.getByName(target));
+        } catch (UnknownHostException e) {
+            throw new IllegalArgumentException(e);
+        }
+    }
+
+    @Override
+    protected Collection<Ban.Ip> getBans() {
+        return getBanService().getIpBans();
+    }
+
+    @Override
+    protected String getTarget(Ban.Ip ban) {
+        return ban.getAddress().toString();
+    }
+
+    @Override
+    public void pardon(String target) {
+        try {
+            getBanService().pardon(InetAddress.getByName(target));
+        } catch (UnknownHostException e) {
+            // Ignore
```

Craftbukkit appears to let it fail silently, so I would stick with that behavior.